conn2res/task.py

```diff
@@ -19,77 +19,120 @@
 from sklearn.metrics import accuracy_score, confusion_matrix, ConfusionMatrixDisplay
 from sklearn.ensemble import RandomForestRegressor
 
-import matplotlib.pyplot as plt
+# import matplotlib.pyplot as plt
 
 
-def check_xy_dims(x,y):
+def check_xy_dims(x, y):
     """
     Check that X,Y have the right dimensions
-    #TODO
+    # TODO
     """
     x_train, x_test = x
     y_train, y_test = y
 
     if ((x_train.ndim == 1) and (x_test.ndim == 1)):
         x_train = x_train[:, np.newaxis]
-        x_test  = x_test[:, np.newaxis]
+        x_test = x_test[:, np.newaxis]
     elif ((x_train.ndim > 2) and (x_test.ndim > 2)):
         x_train = x_train.squeeze()
-        x_test  = x_test.squeeze()
+        x_test = x_test.squeeze()
 
     y_train = y_train.squeeze()
-    y_test  = y_test.squeeze()
+    y_test = y_test.squeeze()
 
     return x_train, x_test, y_train, y_test
 
 
-def regression(x, y, **kwargs):
+def regression(x, y, model=None, **kwargs):
     """
     Regression tasks
-    #TODO
+    # TODO
     """
 
+    # pop variables from kwargs
+    sample_weight_train, sample_weight_test = kwargs.pop(
+        'sample_weight', (None, None))
+
     x_train, x_test = x
     y_train, y_test = y
 
-    model = Ridge(fit_intercept=False, alpha=0.5, **kwargs).fit(x_train, y_train)
-    score = model.score(x_test, y_test)
+    # specify default model
+    if model is None:
+        model = Ridge(fit_intercept=False, alpha=0.5,
+                      **kwargs)
```

Comment:
This default model that I kept for backward compatibility is used inconsistently:

- alpha is set to 0.5 whereas in other instances of ridge regression/classifier alpha is set to 0.0 or 1.0.